plugins/FiberDRCaloSDAction.cpp

```diff
@@ -0,0 +1,245 @@
+        {
+          if (stepTime < ((fTimeStart + static_cast<float>(i) * fTimeStep) * CLHEP::ns))
+            break;
+        }
+
+        return i;
+      }
+
+      bool applyYellowFilter(G4double en)
+      {
+        double energy = (en * 1000000.); // MeV -> eV unit change
+
+        if (energy >= 2.69531)
+          return true; // Photon w/ E > 2.69531 eV filtered
+
+        TF1 *uniform = new TF1("f1", "pol0", 0, 1); // TODO :: Pointer -> change to unique pointer
+        uniform->SetParameter(0, 1);
+        double randVal = uniform->GetRandom(0, 1); // TODO :: Change random engine to DDSim or G4 random!
+        const int N = 24;
+        double graph_X[N] = {1.37760,
+                             1.45864,
+                             1.54980,
+                             1.65312,
+                             1.71013,
+                             1.77120,
+                             1.83680,
+                             1.90745,
+                             1.98375,
+                             2.06640,
+                             2.10143,
+                             2.13766,
+                             2.17516,
+                             2.21400,
+                             2.25426,
+                             2.29600,
+                             2.33932,
+                             2.38431,
+                             2.43106,
+                             2.47968,
+                             2.53029,
+                             2.58300,
+                             2.63796,
+                             2.69531};
+        double graph_Y[N] = {0.903 * 0.903,
+                             0.903 * 0.903,
+                             0.903 * 0.903,
+                             0.903 * 0.903,
+                             0.903 * 0.903,
+                             0.903 * 0.903,
+                             0.902 * 0.902,
+                             0.901 * 0.901,
+                             0.898 * 0.898,
+                             0.895 * 0.895,
+                             0.893 * 0.893,
+                             0.891 * 0.891,
+                             0.888 * 0.888,
+                             0.883 * 0.883,
+                             0.87 * 0.87,
+                             0.838 * 0.838,
+                             0.76 * 0.76,
+                             0.62 * 0.62,
+                             0.488 * 0.488,
+                             0.345 * 0.345,
+                             0.207 * 0.207,
+                             0.083 * 0.083,
+                             0.018 * 0.018,
+                             0.};
+        TGraph *FilterEffGraph = new TGraph(N, graph_X, graph_Y); // TODO :: Pointer -> change to unique pointer
+        double FilterEff = FilterEffGraph->Eval((double)energy);
+
+        delete uniform;
+        delete FilterEffGraph;
+
+        // filter efficiency == probability of photon passing filter
```

Comment:
I still can't tell if this is a pass or deny filter. Does passing the filter mean rejection or acceptance?